torch/distributed/utils.py

```diff
@@ -0,0 +1,56 @@
+import torch
+
+def _parse_remote_device(remote_device: str):
+    r"""
+    Parses the remote device.
+
+    Args:
+        remote_device (str): Device on the destination worker where we'd like to place this module.
+            The format should be one of the following:
+
+                1. "<workername>/<device>", where the device field can be parsed as torch.device type.
+                   E.g., "trainer0/cpu", "trainer0", "ps0/cuda:0".
+                   In addition, the device field can be optional and the default value is "cpu".
+                2. "rank:<rank>/<device>", where <rank> is the rank of the
+                   process and device can be parsed as torch.device type.
+                   E.g., "rank:0/cpu", "rank:0", "rank:0/cuda:0"
+
+    Returns:
+        A workername/rank and a device.
+    """
+
+    PARSE_ERROR = (
+        f"Could not parse remote_device: {remote_device}. The valid format is "
+        "'<workername>/<device>' or 'rank:<rank>/<device>'"
+    )
+
+    fields = remote_device.split("/")
+    if len(fields) == 2:
+        [on, device] = fields
+    elif len(fields) == 1:
+        on = fields[0]
+        device = "cpu"
+    else:
+        raise ValueError(PARSE_ERROR)
+
+    # Since the workername in the input remote device won't be validated until the created remote module is executed,
```

Comment:
This means we expect ShardingSpec to be created before `init_process_group` and `init_rpc`? If so, does it mean users need to do another round of correctness verification after c10d/rpc are initialized? Will it make sense to require that ShardingSpec can only be created after c10d/rpc initialization, so that we can do more correctness checks here?

Reply:
I don't think ShardingSpec creation is tied to `init_process_group` or `init_rpc` and it shouldn't be tied to them. The specification should be independent of the runtime environment such that I can apply the same specification to different training jobs as an example.